Adjust the code introduced with "x86/mm: Make PV linear pagetables
optional" to make it disabled by default, as was in original XSA-240
patch:

    From eeea00c427376c269ffa34587f16fa1182d8c29d Mon Sep 17 00:00:00 2001
    Date: Fri, 22 Sep 2017 11:46:55 +0100
    From: George Dunlap <george.dunlap@citrix.com>
    Subject: [PATCH 2/2] x86/mm: Disable PV linear pagetables by default

    Allowing pagetables to point to other pagetables of the same level
    (often called 'linear pagetables') has been included in Xen since its
    inception.  But it is not used by the most common PV guests (Linux,
    NetBSD, minios), and has been the source of a number of subtle
    reference-counting bugs.

    Add a command-line option to control whether PV linear pagetables are
    allowed (disabled by default).

    Signed-off-by: George Dunlap <george.dunlap@citrix.com>
    Reviewed-by: Andrew Cooper <andrew.cooper3@citrix.com>
    ---
    Changes since v2:
    - s/_/-/; in command-line option
    - Added __read_mostly
    ---
     docs/misc/xen-command-line.markdown | 15 +++++++++++++++
     xen/arch/x86/mm.c                   |  9 +++++++++
     2 files changed, 24 insertions(+)

diff --git a/docs/misc/xen-command-line.markdown b/docs/misc/xen-command-line.markdown
--- a/docs/misc/xen-command-line.markdown	2018-07-16 21:15:46.809000000 +0200
+++ b/docs/misc/xen-command-line.markdown	2018-07-16 21:19:33.532000000 +0200
@@ -1301,10 +1301,7 @@
 ### pv-linear-pt
 > `= <boolean>`
 
-> Default: `true`
-
-Only available if Xen is compiled with CONFIG\_PV\_LINEAR\_PT support
-enabled.
+> Default: `false`
 
 Allow PV guests to have pagetable entries pointing to other pagetables
 of the same level (i.e., allowing L2 PTEs to point to other L2 pages).
diff --git a/xen/arch/x86/mm.c b/xen/arch/x86/mm.c
--- a/xen/arch/x86/mm.c	2018-07-16 21:15:46.819000000 +0200
+++ b/xen/arch/x86/mm.c	2018-07-16 21:20:39.774000000 +0200
@@ -775,7 +775,7 @@
  *     frame if it is mapped by a different root table. This is sufficient and
  *     also necessary to allow validation of a root table mapping itself.
  */
-static bool_t __read_mostly opt_pv_linear_pt = 1;
+static bool_t __read_mostly opt_pv_linear_pt = 0;
 boolean_param("pv-linear-pt", opt_pv_linear_pt);
 
 #define define_get_linear_pagetable(level)                                  \
